src/pages/patientView/PatientViewPage.tsx

```diff
@@ -545,12 +545,17 @@ export default class PatientViewPage extends React.Component<IPatientViewPagePro
                     </MSKTab>
                     )}
 
-                    <MSKTab key={7} id="trialMatchTab" linkText="Matched Trials"
-                            hide={!this.shouldShowTrialMatch(patientViewPageStore)}>
-                        <TrialMatchTable sampleManager={sampleManager}
-                                         detailedTrialMatches={patientViewPageStore.detailedTrialMatches.result}
-                                         containerWidth={WindowStore.size.width-20}/>
-                    </MSKTab>
+                    {
+                        this.shouldShowTrialMatch(patientViewPageStore) && (
```

Comment:
@victoria34 i should have caught this earlier but the way mobx promise works, the promise gets called even if component is hidden.  This change ensures that it is not invoked unless shouldShow is true

Reply:
@alisman Sorry, I didn't notice that. Will be careful next time. Thanks for letting me know.